eth/eventsyncer/event_syncer.go

```diff
@@ -80,6 +85,29 @@ func (es *EventSyncer) Healthy(ctx context.Context) error {
 	if time.Since(es.lastProcessedBlockChange) > es.stalenessThreshold {
 		return fmt.Errorf("syncing is stuck at block %d", lastProcessedBlock.Uint64())
 	}
+
+	highestSeenBlock, found, err := es.nodeStorage.GetHighestSeenBlock(nil)
+	if err != nil {
+		return fmt.Errorf("failed to read last processed block: %w", err)
+	}
+	if !found || highestSeenBlock == nil || highestSeenBlock.Uint64() == 0 {
+		return fmt.Errorf("last seen block is not set")
+	}
+
+	return es.blockBelowThreashold(ctx, highestSeenBlock)
+}
+
+func (es *EventSyncer) blockBelowThreashold(ctx context.Context, block *big.Int) error {
+	header, err := es.executionClient.HeaderByNumber(ctx, block)
+	if err != nil {
+		return fmt.Errorf("failed to get header for block %d: %w", block, err)
+	}
+
+	// #nosec G115
+	if header.Time != 0 && header.Time < uint64(time.Now().Add(-es.stalenessThreshold).Unix()) {
```

Comment:
if header.Time is 0 something is wrong we should treat it as block is too old imo (unless we find out its a legit response and handle it differently)